packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.Props.ts

```diff
@@ -261,7 +261,7 @@ export interface IContextualMenuItem {
    * Method to custom render this menu item
    * @defaultvalue undefined
    */
-  onRender?: (item: any) => React.ReactNode;
+  onRender?: (item: any, index?: number, hasCheckmarks?: boolean, hasIcons?: boolean) => React.ReactNode;
```

Comment:
I am ok with making this modification to unblock a core scenario, but we need to fix the onRender to be proper. That way the caller can pass in all the IButtonProp settings and we don't have to change the contract every time.